be/src/olap/rowset/segment_v2/segment_iterator.cpp

```diff
@@ -466,8 +466,12 @@ Status SegmentIterator::_execute_predicates_except_leafnode_of_andnode(vectorize
     } else if (_is_literal_node(node_type)) {
         auto v_literal_expr = dynamic_cast<doris::vectorized::VLiteral*>(expr);
         _column_predicate_info->query_value = v_literal_expr->value();
-    } else if (node_type == TExprNodeType::BINARY_PRED) {
```

Comment:
another " else if " is better

Reply:
The only difference between `BINARY_PRED` and `MATCH_PRED` branch is `query_op` is different, use another "else if" will be some duplicate code.
Specify `MATCH_PRED`'s query_op as "match" not `expr->fn().name.function_name` in order to unify "match_any", "match_all" and so on to "match"